Network/Haskoin/Node/SpvBlockChain.hs

```diff
@@ -247,12 +247,20 @@ headerSync resource locType hStopM = do
 processMerkleBlocks :: (HeaderTree m, MonadLogger m, MonadIO m)
                     => DwnId -> [DecodedMerkleBlock] -> StateT SpvSession m ()
 
-processMerkleBlocks did [] = do
-    $(logError) $ format $ "Got a completed merkle job with an empty merkle list"
-    modify $ \s -> s{ merkleId = Nothing }
-    -- We continue the merkle download because we didn't have any transactions
-    continueMerkleDownload
-    checkSynced
+processMerkleBlocks did [] = gets merkleId >>= \mid -> case mid of
+    Just (expId, _) -> if did == expId
+        then do
+            $(logError) $ format $ 
```

Comment:
We should not receive an empty merkle block job from a peer as we only create non-empty merkle download jobs for peers. I think we should never hit this computation branch but I log an error here just in case we do get here in an unexpected way.